lib/gitlab/ci/templates/pipeline/NpmJestCoveragePipeline.yml

```diff
@@ -0,0 +1,57 @@
+stages:
+  - build
+  - test
+  - deploy
+
+build:
+  image: node:16
+  stage: build
+  script:
+    - npm install
+    - npm run build
+  cache:
+    paths:
+      - node_modules/
+  artifacts:
+    expire_in: 1 days
+    when: on_success
+    paths:
+      - node_modules/
+
+
+# Run Tests
+test:
+  image: node:16
+  stage: test
+  coverage: '/Lines\s*:\s*(\d+.?\d*)%/'
+  dependencies:
+    - build
+  script:
+    - npm run test:ci
```

Comment:
Does NPM support any custom flags? We should probably add a variable here to enable consumers to customize this.